prog/dftb+/lib_dftb/etemp.F90

```diff
@@ -464,8 +514,10 @@ subroutine electronFill(Eband, filling, TS, E0, Ef, eigenvals, kT, distrib, kWei
 #:else
             filling(j, i, iSpin) = 1.0_dp / (1.0_dp + exp(x))
 #:endif
-            if (filling(j, i, iSpin) <= elecTol) then
+            if (tNonAufbau .and. j/=1 .and. ((filling(j, i, iSpin)+filling(j-1, i, iSpin))) <= elecTol) then
```

Comment:
When j==1 in the loop, the j-1 term goes below the lower limit of the filling array (compiling with DEBUG := 2 in the make.config file finds this).

Reply:
the slightly hacky fix is filling(j-1, i, iSpin) =>  filling(max(j-1,1), 1, iSpin)
It might be slightly clearer to split the test as 
if (tNonAufbau .and. j/=1) then
  if ( filling(j, i, iSpin)+filling(j-1, i, iSpin) <= elecTol) then

Reply:
Either change works equally well, do you have a preference? I prefer the more compact hacky fix, but either way is fine by me!

Reply:
The max() form is OK for me (Bálint may disagree...), but it doesn't cover the 1 electron case. it might be worth testing for the 1 electron case in initprogram.F90 and stopping with an error if it doesn't fill correctly in several places.

Reply:
You make a good point, we have assumed a closed shell ground state system with an even number of electrons, but we have not coded that assumption in yet. I will get right on that! I will enforce that restriction in initprogram.F90